DM-39789: EfdClient: accept non-utc astropy Times

r-owen opened the pull request.

jonathansick reviewed (commented):
Could you split the "EfdClient: accept non-utc astropy Times" between what is necessary for the PR and what is part of the black/isort reformat? Thanks.

r-owen commented:
Good suggestion @jonathansick. I redid the changes, starting with the black and isort much earlier (so I hope you decide you can live with them!) and now it's all tidy. (I applied black early so that I didn't have to disable automatic formatting in my editor).

I also added pre-commit config and updated the black and isort config so it should all be self-consistent.

afausti reviewed (approved):
Thanks for doing this change and fixing the tests. I agree that converting timestamps to UTC internally is better than raising an error. Looks great.
